### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by codedit334**
# Hi Reviewer 👋,

This Project follows design by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

### Technical requirements
  - Uses Postgres as your database.
  - Uses devise for authentication.
  - Validates all user input to make sure that anyone with bad intentions cannot compromise your app.
  - Deployed and accessible online.
  - The database schema should reflect the provided one.

### Interactions
- Splash screen

  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages

  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 
- Home page (categories page)

  - When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
 
  - Transactions page

  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
 
- "Add a new category" page

  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
  
 - "Add a new transaction" page

  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
  
# ✋

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <div class="flex justify-between items-center h-11 bg-[#3778c2] text-white mb-10"> 
+    <div class="ml-4 cursor-pointer">
+      <%= link_to "<", unauthenticated_root_path %>
+    </div>
+    <h2 class="uppercase">
+      Register
+    </h2>
+    <div class="actions mr-4 cursor-pointer">
+      <%= f.submit "Next" %>
```
Comment (ShadyShawkat) [inline app/views/devise/registrations/new.html.erb:11]: - [ ] Kindly make sure that all forms have the same style format as the original design.

| Your design | Original design |
| --- | --- |
| image | image |

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ <div class="flex justify-between items-center h-11 bg-[#3778c2] text-white"> 
+    <div class="ml-4 cursor-pointer">
+      <%= link_to "<", authenticated_root_path %>
+    </div>
+    <h2 class="uppercase">
+      Categories
+    </h2>
+    <div class="actions mr-4 cursor-pointer text-2xl">
+      <%= link_to "+", new_category_path %>
+    </div>
```
Comment (ShadyShawkat) [inline app/views/categories/index.html.erb]: - [ ] Please implement the "Add" button to be at the bottom of the design to better match the original design.

image

File: app/views/its_transactions/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<div class="flex justify-between items-center h-11 bg-[#3778c2] text-white"> 
+    <div class="ml-4 cursor-pointer">
+      <%= link_to "<", authenticated_root_path %>
+    </div>
+    <h2 class="uppercase">
+      Trasactions
+    </h2>
+    <div class="actions mr-4 cursor-pointer text-2xl">
+      <%= link_to "+", new_category_its_transaction_path(@category) %>
+    </div>
```
Comment (ShadyShawkat) [inline app/views/its_transactions/index.html.erb]: - [ ] Please implement the "Add" button to be at the bottom of the design to better match the original design.

image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @codedit334, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @codedit334 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Use of `devise` gem for authentication ✔️ 
- Application design is identical to the template provided ✔️ 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**